frontend/src/components/ChatContainer/ChatWindow/ChatHeader/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Box,
+  IconButton,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  makeStyles,
+  Tooltip,
+} from '@material-ui/core';
+import ReplayIcon from '@material-ui/icons/Replay';
+import RemoveIcon from '@material-ui/icons/Remove';
+import CloseIcon from '@material-ui/icons/Close';
+import Avatar from 'components/Avatar';
+import { ChatWindowStoreContext } from 'contexts';
+import { observer } from 'mobx-react-lite';
+import { chatContainerStore } from 'stores';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    width: '100%',
+  },
+  icon: {
+    height: theme.spacing(2),
+    width: theme.spacing(2),
+  },
+  infoWindow: {
+    maxWidth: '70%',
+  },
+}));
+
+const ChatHeader = (props) => {
+  const { chatId } = props;
+  const classes = useStyles();
+  const chatWindowStore = useContext(ChatWindowStoreContext);
+  const { name, setWindowMinimized, reconnect } = chatWindowStore;
+
+  const handleMinimize = () => {
+    setWindowMinimized(true);
```

Comment:
'this' is undefined when calling this funtion using object destructuring
but works fine if called on chatWidnowStore(chatWindowStore.setWindowMinimized(ture)).

Please check if 'this' is bound or not for setWindowMinimized.
Also even if it is bound using action.bound, are we allowed to use object destructuring and call the function?

Reply:
Classes are just objects in javascript so we can freely destructure.